src/debug.cc

```diff
@@ -975,22 +974,25 @@ SyslogPriority(const DebugMessageHeader &header)
            (header.level == 0 ? LOG_WARNING : LOG_NOTICE);
 }
 
+bool
+SyslogChannel::shouldWrite(const DebugMessageHeader &header) const
+{
+    return header.forceAlert ||
+        header.level == DBG_CRITICAL ||
+        (Debug::log_syslog && header.level <= DBG_IMPORTANT);
+}
+
 void
 SyslogChannel::log(const DebugMessageHeader &header, const std::string &body)
 {
     if (header.recordNumber <= lastWrittenRecordNumber)
         return;
 
-    if (earlyMessages)
-        return (void)saveMessage(header, body);
-
-    if (!header.forceAlert) {
-        if (header.level > DBG_IMPORTANT)
-            return;
+    if (!shouldWrite(header))
+        return saveMessage(header, body);
 
-        if (!Debug::log_syslog)
-            return;
-    }
+    // Do not stopEarlyMessageCollection() here: The already saved earlier
+    // messages are doomed, but future early messages still have a chance.
```

Comment:
```suggestion
    // messages are doomed, but future early DBG_IMPORTANT messages should
    // be saved in case Debug::log_syslog becomes true later.
```